### DISCUSSION TITLE
Issue with launching with Docker

### DISCUSSION THREAD

**Discussion by bclaudio08** [Mirobit/bitcoin-node-manager#65] [Q&A]
Hi,

I am trying to launch via Docker and `http://server-ip:8000/` but when I do I get two errors:

`Warning: session_start(): Cannot start session when headers already sent in /var/www/html/index.php on line 48`

`Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/src/Config.php:60) in /var/www/html/index.php on line 72`

Tried Googling around to find a fix but all my attempts were unsuccessful. Any direction or assistance is greatly appreciated.

Thanks

**Comment by Mirobit** [comment]
Does it work without docker?

**Reply by k0gen** [reply]
It is working with Docker for sure - that is exactly how I use it. But from time to time I get this on "Main" interface load and sometimes on "Blocks" tab:

<img width="1632" alt="Screenshot 2023-08-03 at 9 39 10 AM" src="https://github.com/Mirobit/bitcoin-node-manager/assets/3606313/895e8988-43a2-4966-aa22-d35db5c49e67">

I'm using alpine + nginx + php-fpm

Update: Those warnings are no longer a thing when using alpine 3.18 and the latest version of php 8.2.8 🎉

Update2: I suppose I spoke too soon. After leaving the service running for some time, I encountered this issue upon opening the BNM:
![Screenshot 2023-08-05 at 15-37-35 Home - Bitcoin Node Manager](https://github.com/Mirobit/bitcoin-node-manager/assets/3606313/62861df0-dd50-48f7-9ddc-3e202f5d72a5)
When I tried to refresh the page, everything is back to normal.

**Comment by k0gen** [comment]
These warnings and deprecated messages are popping up on the Main page from time to time:
```
Warning: Undefined array key "countryCode" in /var/www/bitcoin-node-manager/src/Utility.php on line 558

Deprecated: preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 160

Warning: Undefined array key "country" in /var/www/bitcoin-node-manager/src/Utility.php on line 559

Deprecated: substr(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 168

Warning: Undefined array key "isp" in /var/www/bitcoin-node-manager/src/Utility.php on line 560

Deprecated: substr(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 168

Warning: Undefined array key "countryCode" in /var/www/bitcoin-node-manager/src/Utility.php on line 558

Deprecated: preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 160

Warning: Undefined array key "country" in /var/www/bitcoin-node-manager/src/Utility.php on line 559

Deprecated: substr(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 168

Warning: Undefined array key "isp" in /var/www/bitcoin-node-manager/src/Utility.php on line 560

Deprecated: substr(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 168

Warning: Undefined array key "countryCode" in /var/www/bitcoin-node-manager/src/Utility.php on line 558

Deprecated: preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 160

Warning: Undefined array key "country" in /var/www/bitcoin-node-manager/src/Utility.php on line 559

Deprecated: substr(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 168

Warning: Undefined array key "isp" in /var/www/bitcoin-node-manager/src/Utility.php on line 560

Deprecated: substr(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/bitcoin-node-manager/src/Utility.php on line 168
```
I'm trying to figure out why this is happening, and it exhibits very inconsistent behavior, making it hard for me to replicate. Does anyone have an idea what is going on or a solution for this??

**Comment by Mirobit** [comment]
I wonder if this related to your tor usage. I am wondering about the 4 Unknown Peers. I never have these on my server.
Do you have peers that you are only connected to via TOR?

**Reply by k0gen** [reply]
Yes, I do.
I have also noticed that those long names do brake the layout.
<img width="1344" alt="Screenshot 2023-08-10 at 13 49 35" src="https://github.com/Mirobit/bitcoin-node-manager/assets/3606313/d1d3b50c-7e41-43f8-8609-bc34bbf1f305">

**Reply by Mirobit** [reply]
Okay, that helps a lot. 
It is whitelisted peers and onions peers that cause the exceptions. There seems to be a bug with caching the geo information for whitelisted peers. Getting the geo data for tor peers can't work, but BNM still tries. Plus tor addresses need to be shortened.
Thanks for providing the screenshot. I will fix these three issues in the next week.
